cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -150,246 +154,253 @@ public void onFailure(int statusCode, String response, Throwable throwable) {
                 continue;
             }
 
             if (banner.getTriggerType() == BannerTriggerType.Conditions) {
-            	boolean triggers = false;
-            	for (BannerTrigger trigger : banner.getTriggers()) {
-            		boolean triggerTrue = false;
-            		for (BannerTriggerCondition condition : trigger.getConditions()) {
-            			boolean conditionTrue = false;
-            			if (condition.getType() != null) {
-							if (condition.getType().equals(BannerTriggerConditionType.Duration)) {
-								banner.setDelaySeconds(condition.getSeconds());
-								conditionTrue = true;
-							}
-							if (condition.getType().equals(BannerTriggerConditionType.Sessions)) {
-								if (condition.getRelation().equals("lt")) {
-									conditionTrue = sessions < condition.getSessions();
-								} else {
-									conditionTrue = sessions > condition.getSessions();
-								}
-							}
-							if (condition.getType().equals(BannerTriggerConditionType.Event)) {
-								String event = events.get(condition.getKey());
-								conditionTrue = event != null && event.equals(condition.getValue());
-							}
-						}
-
-						if (conditionTrue) {
-							triggerTrue = true;
-							break;
-						}
-					}
-
-            		if (triggerTrue) {
-						triggers = true;
-						break;
-					}
-				}
-
-            	if (!triggers) {
-					Log.d(TAG, "Skipping Banner because: Trigger not satisfied " + sessions);
-            		continue;
-				}
-			}
+                boolean triggers = false;
+                for (BannerTrigger trigger : banner.getTriggers()) {
+                    boolean triggerTrue = false;
+                    for (BannerTriggerCondition condition : trigger.getConditions()) {
+                        boolean conditionTrue = false;
+                        if (condition.getType() != null) {
+                            if (condition.getType().equals(BannerTriggerConditionType.Duration)) {
+                                banner.setDelaySeconds(condition.getSeconds());
+                                conditionTrue = true;
+                            }
+                            if (condition.getType().equals(BannerTriggerConditionType.Sessions)) {
+                                if (condition.getRelation().equals("lt")) {
+                                    conditionTrue = sessions < condition.getSessions();
+                                } else {
+                                    conditionTrue = sessions > condition.getSessions();
+                                }
+                            }
+                            if (condition.getType().equals(BannerTriggerConditionType.Event)) {
+                                String event = events.get(condition.getKey());
+                                conditionTrue = event != null && event.equals(condition.getValue());
+                            }
+                        }
+
+                        if (conditionTrue) {
```

Comment:
(as noted above with proper termination statement)

Reply:
We'll rework in SDK 2.0